eras/allegra/impl/cardano-ledger-allegra.cabal

```
@@ -46,7 +46,7 @@ library
         -Wunused-packages
 
     build-depends:
-        base >=4.12 && <4.17,
+        base >=4.12 && <5,
```

We do not support ghc-9.4 yet. When I said we need to revert back all removed constraints on base I meant revert them literally
```suggestion
        base >=4.12 && <4.17,
```